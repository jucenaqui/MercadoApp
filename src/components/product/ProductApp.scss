.product-list {
    background-color: white;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 80%;
    margin: 20px auto;
}