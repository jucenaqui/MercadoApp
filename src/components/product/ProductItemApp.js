import React from 'react';

import './ProductItemApp.scss';

export const ProductItemApp = ( { product}) => {
    
    return (
        <>
      
            <div className="product-item">
                <img className="producto-item-img" src={product.thumbnail}  alt={product.title}/>
                <div className="product-item-description"> 
                <h3>$ {product.price}</h3>   
                <p>{product.title}</p>
                </div>
            </div>
        </>
    )
}
