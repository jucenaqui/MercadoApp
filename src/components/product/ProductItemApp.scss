.product-item {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 10px 20px;
    box-shadow: 0px 1px 5px #f3f2f2;

    img {
        margin: 10px;
        border-radius: 3px;
        width: 100px;
    }
}

.product-item-description {
    align-items: flex-start;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    padding-top: 20px;

    p {
        margin: 0 !important;
        width: 90%;
    }

    h3 {
        margin: 10px 0;
    }
}

@media screen and (max-width: '768px') {
    .product-item {
        flex-direction: column;
        justify-content: center;
        align-items: center;

        img {
            width: 150px;
        }
    }

    .product-item-description {
        align-items: center;

        p {
            width: 100%;
            padding: 5px 20px;
        }
    }
}