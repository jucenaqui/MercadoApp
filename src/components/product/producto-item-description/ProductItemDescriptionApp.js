import React from 'react';

import PropTypes from 'prop-types';

import './ProductItemDescriptionApp.scss';

export const ProductItemDescriptionApp = ( { product, categories } ) => {
    
    return (
        <>
            <p className="product-category"> 
                { 
                    categories && categories.map(item => item.name).join(' > ')
                } 
            </p>
            <div className="product-item animate__animated animate__fadeIn">
                <div className="container-product">
                    <div className="container-product-item">
                        <img className="producto-item-img" src={product.picture}  alt={product.title}/>
                        <div className="product-item-params">
                            <span className="producto-item-condition">{ product?.condition === 'new' ? 'Nuevo': product?.condition } - {product?.sold_quantity } Vendidos </span>
                            <p className="producto-item-title">{product.title}</p>
                            <h2>$ {product.price?.amount}</h2>   
                            <button className="producto-item-button">Comprar</button>
                        </div>
                    </div>
                    <div className="product-item-description-app"> 
                        <h2 className="description-title">Descrici√≥n del producto</h2>
                        <p className="description">{product.description}</p>
                    </div>
                </div>
            </div>
        </>
    )
}

ProductItemDescriptionApp.propTypes = {
    product: PropTypes.object.isRequired,
    categories: PropTypes.array.isRequired
}
