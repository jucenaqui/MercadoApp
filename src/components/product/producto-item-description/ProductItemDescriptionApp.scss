.product-item {
    display: flex;
    flex-direction: column;
    padding: 0 30px;

    .container-product {
        background-color: white;
        padding: 10px;
        width: 80%;
        margin: 0 auto;
    }
    .container-product-item {
        display: flex;
        justify-content: space-evenly;
        padding: 20px;
        border-radius: 4px;

        img {
            margin: 10px;
            border-radius: 3px;
            width: 350px;
        }

        .product-item-params {
            margin-top: 10px;

            .producto-item-condition {
                color: gray;
            }

            .producto-item-title {
                font-size: 1.2rem;
                width: 200px;
            }

            .producto-item-button {
                background-color: #0e9fec;
                border: none;
                border-radius: 3px;
                color: white;
                font-size: 1rem;
                padding: 8px 20px;
                margin-top: 1.5rem;
                width: 100%;
            }

            h2 {
                margin: 5px 0px;
            }
        }

    }
}

.product-item-description-app {
    
    width: 100%;
    padding: 10px 20px;
    box-sizing: border-box;
    p {
        width: 80%;
        padding: 10px;
    }

    h2 {
        margin: 10px 0;
        margin-left: 10px;
    }
}

@media screen and (max-width: '768px') {
    .container-product-item {
        flex-direction: column;
        align-items: center;

        img {
            width: 300px;
        }
    }

    .product-item-params {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;

        .producto-item-title {
            width: 100%;
            text-align: center;
        }

        .producto-item-button {
            width: 80%;
        }
    }

    .product-item-description-app {
    
        h2 {
            text-align: center;
        }

        p {
            width: 90%;
            text-align: justify;
        }

    }
}