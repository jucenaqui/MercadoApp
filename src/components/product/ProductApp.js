import React from 'react';
import {ProductItemApp} from './ProductItemApp';

import './ProductApp.scss';

const ProductApp = ( { products } ) => {

    return (
     <>
        <ol className="product-list">
            {
                products.map((product)=> (<ProductItemApp key={product.id} product={product} />))
            }
        </ol>
     </>
    )
}

export default ProductApp;