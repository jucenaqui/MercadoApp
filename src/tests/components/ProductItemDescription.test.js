import { shallow } from "enzyme"
import { ProductItemDescriptionApp } from "../../components/product/producto-item-description/ProductItemDescriptionApp";
import { productMuck, categoriesMuck } from '../mocks/product';

describe('Pruebas <ProductItemDescription />', () => {
    
    const product = productMuck;
    const categories = categoriesMuck;
    const wrapper = shallow(<ProductItemDescriptionApp 
                            product={ product} categories={ categories } />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a <p> with the title', () => { 
        const title = 'Heladera';
        const p = wrapper.find('p.producto-item-title');
        expect(p.text().trim()).toBe(title);
    });

    test('should have a <h3> with a price', () => {
        const price = `$ ${product.price.amount}`;
        const div = wrapper.find('div.product-item-params');
        const h2 = div.props().children.filter(i=>i.type==='h2')[0].props.children;
        expect(h2.join('')).toBe(price);
    });
    
    test('should have a img with a src and alt attibutes', () => {
        const imgSrc = product.picture;
        const imgAlt = product.title;
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(imgSrc)
        expect(img.prop('alt')).toBe(imgAlt);
        expect(img.prop('src').includes('http://')).toBe(true);
    });

    test('should have a h2 with title description', () => {
        const description = 'DescriciÃ³n del producto';
        const h2 = wrapper.find('h2.description-title');
        expect(h2.text().trim()).toBe(description);
    });

    test('should have a p with a description', () => {
        const description = productMuck.description;
        const p = wrapper.find('p.description');
        expect(p.text().trim()).toBe(description);
    });

    test('should have a p with a category string', () => {
        const categories = categoriesMuck.map(i=>i.name).join(' > ');
        const p = wrapper.find('p.product-category');
        expect(p.text().trim()).toBe(categories);
    });
    
    

})
