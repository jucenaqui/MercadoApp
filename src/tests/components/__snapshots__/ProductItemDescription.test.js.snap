// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <ProductItemDescription /> should show the component succesfull 1`] = `
<Fragment>
  <p
    className="product-category"
  >
    Electrodomésticos &gt; Refrigeración &gt; Refrigeradores
  </p>
  <div
    className="product-item animate__animated animate__fadeIn"
  >
    <div
      className="container-product"
    >
      <div
        className="container-product-item"
      >
        <img
          alt="Heladera"
          className="producto-item-img"
          src="http://http2.mlstatic.com/D_965028-MLA32542008594_102019-I.jpg"
        />
        <div
          className="product-item-params"
        >
          <span
            className="producto-item-condition"
          >
            Nuevo
             - 
             Vendidos 
          </span>
          <p
            className="producto-item-title"
          >
            Heladera
          </p>
          <h2>
            $ 
            47200
          </h2>
          <button
            className="producto-item-button"
          >
            Comprar
          </button>
        </div>
      </div>
      <div
        className="product-item-description-app"
      >
        <h2
          className="description-title"
        >
          Descrición del producto
        </h2>
        <p
          className="description"
        >
          la description
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
